### ISSUE TITLE
many crashes in new Release r234.2  (OSX) (pick by unique ID may be the problem)

### ISSUE THREAD

**Issue by bubbblebub** [Scirra/Construct-3-bugs#4545]
## Problem description
in the new Release r234.2 Construct 3 crashes very often (also in r234).
 
It happens most when I copy paste events in the event sheet, but I think it is releated with a problem with the "Pick by unique ID" Dialogue who is some times bugy. Please take a look at the Screenshot - there is two times the same "unique ID". After I closed this Dialogue it crashed. I think I only experience crashes when I copy paste Events with this Unique ID event included, not sure.

![Bildschirmfoto 2021-02-07 um 09 57 23](https://user-images.githubusercontent.com/35291800/107141883-e0258e80-692b-11eb-880a-1c1acea17387.png)

Ufortunately I was not able to reproduce it on purpose, I do not have a minimal reproduction project- I can only send you the Dialogue from the "oops something went wrong" report and the Screenshot from the "Pick by unique ID" Dialogue. 
It just happens from time to time. I had the situation where it chrashed two times in 10 minutes, often just once, sometimes I am able to work for hours. I think it happens more in the beginning of a session, when it works for an hour I never had a crash afterwards.
First I thought it is a chrome bug, then I changed to Firefox. But now I just had the first crash on Firefox.

## Steps to reproduce

1.Not able to reproduce it
2.
3.

## More details

this is an error Report from a copy paste of events crash (see below for the error report after I closed the screenshot dialog):
Error report information

Type: unhandled rejection
Reason: Error: unexpected type @ d.i@https://editor.construct.net/r234-2/main.js:219:490 N@https://editor.construct.net/r234-2/projectResources.js:2371:270 E@https://editor.construct.net/r234-2/projectResources.js:2379:133 C@https://editor.construct.net/r234-2/projectResources.js:2378:474 F@https://editor.construct.net/r234-2/projectResources.js:2378:375 F/<@https://editor.construct.net/r234-2/projectResources.js:2378:432
Stack: d.i@https://editor.construct.net/r234-2/main.js:219:490 N@https://editor.construct.net/r234-2/projectResources.js:2371:270 E@https://editor.construct.net/r234-2/projectResources.js:2379:133 C@https://editor.construct.net/r234-2/projectResources.js:2378:474 F@https://editor.construct.net/r234-2/projectResources.js:2378:375 F/<@https://editor.construct.net/r234-2/projectResources.js:2378:432
Construct 3 version: r234.2
URL: https://editor.construct.net/
Date: Sun Feb 07 2021 09:25:49 GMT+0100 (Mitteleuropäische Normalzeit)
Uptime: 1132.8 s
Platform information

Browser: Firefox
Browser version: 85.0
Browser engine: Gecko
Browser architecture: (unknown)
Context: browser
Operating system: Mac OS X
Operating system version: 10
Operating system architecture: (unknown)
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: (unavailable)
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:85.0) Gecko/20100101 Firefox/85.0
C3 release: r234.2 (stable)
Language setting: en-US
WebGL information

Version string: WebGL 2.0
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: ATI Technologies Inc.
Renderer: AMD Radeon Pro 560 OpenGL Engine
Major performance caveat: no
Maximum texture size: 8192
Point size range: 1 to 2048
Extensions: EXT_color_buffer_float, EXT_float_blend, EXT_texture_compression_rgtc, EXT_texture_filter_anisotropic, OES_texture_float_linear, WEBGL_compressed_texture_s3tc, WEBGL_compressed_texture_s3tc_srgb, WEBGL_debug_renderer_info, WEBGL_debug_shaders, WEBGL_lose_context

Here is the Report from the moment when I closed the screenshot Dialog (this time in Chrome):

Error report information
Type: unhandled rejection
Reason: Error: unexpected type @ TypeError: unexpected type at Function.d.i (https://editor.construct.net/r234-2/main.js:219:490) at d.g.J.XMd.N (https://editor.construct.net/r234-2/projectResources.js:2371:270) at E (https://editor.construct.net/r234-2/projectResources.js:2379:133) at C (https://editor.construct.net/r234-2/projectResources.js:2378:474) at F (https://editor.construct.net/r234-2/projectResources.js:2378:375) at https://editor.construct.net/r234-2/projectResources.js:2378:432
Stack: TypeError: unexpected type at Function.d.i (https://editor.construct.net/r234-2/main.js:219:490) at d.g.J.XMd.N (https://editor.construct.net/r234-2/projectResources.js:2371:270) at E (https://editor.construct.net/r234-2/projectResources.js:2379:133) at C (https://editor.construct.net/r234-2/projectResources.js:2378:474) at F (https://editor.construct.net/r234-2/projectResources.js:2378:375) at https://editor.construct.net/r234-2/projectResources.js:2378:432
Construct 3 version: r234.2
URL: https://editor.construct.net/
Date: Sun Feb 07 2021 09:57:34 GMT+0100 (Mitteleuropäische Normalzeit)
Uptime: 225.4 s

Platform information
Browser: Chrome
Browser version: 88.0.4324.150
Browser engine: Chromium
Browser architecture: (unknown)
Context: browser
Operating system: Mac OS X
Operating system version: 10.15_7
Operating system architecture: (unknown)
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.150 Safari/537.36
C3 release: r234.2 (stable)
Language setting: en-US

WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: ATI Technologies Inc.
Renderer: AMD Radeon Pro 560 OpenGL Engine
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 2048
Extensions: EXT_color_buffer_float, EXT_disjoint_timer_query_webgl2, EXT_float_blend, EXT_texture_compression_rgtc, EXT_texture_filter_anisotropic, EXT_texture_norm16, OES_texture_float_linear, WEBGL_compressed_texture_s3tc, WEBGL_compressed_texture_s3tc_srgb, WEBGL_debug_renderer_info, WEBGL_debug_shaders, WEBGL_lose_context, WEBGL_multi_draw

**Affected browsers/platforms:** 
I tested in chrome and Firefox, both times I had crashes

**First affected release:** 
r 234
## System details

<details><summary>View details</summary>

Platform information

Browser: Firefox
Browser version: 85.0
Browser engine: Gecko
Browser architecture: (unknown)
Context: browser
Operating system: Mac OS X
Operating system version: 10
Operating system architecture: (unknown)
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: (unavailable)
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:85.0) Gecko/20100101 Firefox/85.0
C3 release: r234.2 (stable)
Language setting: en-US
Local storage

Storage quota (approx): 2 gb
Storage usage (approx): 9.9 mb (0.5%)
Persistant storage: No
Browser support notes

This list contains missing features that are not required, but could improve performance or user experience if supported.

    Rendering multiple on-screen Layout Views is slow in Firefox due to bug 1163426
    The <dialog> element is not supported. A polyfill is in use.
    The Clipboard API is not supported. Some clipboard features may be unavailable.
    ImageBitmapOptions is not supported. Texture loading performance may be degraded.
    Determining input device capabilities is not supported.
    image-set() is not supported. Mouse cursors may appear with reduced quality.

WebGL information

Version string: WebGL 2.0
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: ATI Technologies Inc.
Renderer: AMD Radeon Pro 560 OpenGL Engine
Major performance caveat: no
Maximum texture size: 8192
Point size range: 1 to 2048
Extensions:

    EXT_color_buffer_float
    EXT_float_blend
    EXT_texture_compression_rgtc
    EXT_texture_filter_anisotropic
    OES_texture_float_linear
    WEBGL_compressed_texture_s3tc
    WEBGL_compressed_texture_s3tc_srgb
    WEBGL_debug_renderer_info
    WEBGL_debug_shaders
    WEBGL_lose_context

Audio information

System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

    WebM Opus (audio/webm; codecs=opus)
    Ogg Opus (audio/ogg; codecs=opus)
    WebM Vorbis (audio/webm; codecs=vorbis)
    Ogg Vorbis (audio/ogg; codecs=vorbis)
    MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
    MP3 (audio/mpeg)
    FLAC (audio/flac)
    PCM WAV (audio/wav; codecs=1)

Supported encode formats:

    WebM Opus (audio/webm; codecs=opus)
    Ogg Opus (audio/ogg; codecs=opus)

Video information

Supported decode formats:

    WebM AV1 (video/webm; codecs=av01.0.00M.08)
    MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
    WebM VP9 (video/webm; codecs=vp9)
    WebM VP8 (video/webm; codecs=vp8)
    Ogg Theora (video/ogg; codecs=theora)
    H.264 (video/mp4; codecs=avc1.42E01E)

Supported encode formats:

    WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by bubbblebub** [comment]
I just had several more crashes, I guess that this are probably two problems- the buggy "pick by unique ID" window is sepparate from the many crashes I have. I just made a new project, added a sprite, and when I closed the add sprite dialoge it crashed. So it is not related to my project. 

I have the impression that it crashes more often when I use the Pen fom my Wacom Cintiq, when I use the trackpad of my mac it seems more stabe. This may be a hint..
